server/middleware/getPrisonerDataMiddleware.ts

```diff
@@ -10,7 +10,14 @@ export default function getPrisonerData(services: Services, options: { minimal?:
   return async (req, res, next) => {
     try {
       const prisonerSearchClient = services.dataAccess.prisonerSearchApiClientBuilder(req.middleware.clientToken)
-      const prisonerData: Prisoner = await prisonerSearchClient.getPrisonerDetails(req.params.prisonerNumber)
+      let prisonerNumber = ''
+      if (req.params?.prisonerNumber) {
+        prisonerNumber = req.params.prisonerNumber
+      } else if (req.query?.prisonerNumber) {
+        prisonerNumber = req.query.prisonerNumber as string
+      }
+
+      const prisonerData: Prisoner = await prisonerSearchClient.getPrisonerDetails(prisonerNumber)
```

Comment:
Could skip calling the endpoint if `prisonerNumber` is still `''` as it's not going to give you anything

Reply:
you're right i'll add that